### ISSUE TITLE
Split of microG core

### ISSUE THREAD

**Issue by Lanchon** [microg/android_packages_apps_GmsCore#258]
hi,

i want to stop google from running privileged code on my device. however, i'm ok with running google code if sandboxed. in fact, i need to. i can't have a phone that cannot run hangouts for instance, because i need voip support to be able to work.

i've posted on the hangouts issue (which looks simple enough, apparently a missing SSL provider), but the microG project does not seem to be prioritizing this: https://github.com/microg/android_packages_apps_GmsCore/issues/72

microG reimplements the google account. but in my view, the google account can be considered part of the google client apps that i am willing to run (as long as they are sandboxed). if i am using google client apps (gmail, hangouts, etc) i might as well also use google's implementation of their account. in fact, issues with microG's version of the google account might crop up later with client app updates, and using google's code instead is probably safer in this regard.

but for many reasons, microG is doing the right thing in reimplementing the google account; i won't argue this. my position is that not everybody using microG is interested in this functionality.

because of the hangouts issue, that has been open for a year without progress, i'll have to abandon microG unfortunately.

fortunately, google's own account manager seems to work with just one privileged permission. see:
http://kindlefireforkid.com/how-to-setup-a-google-account-on-amazon-fire-tablet/

i'd like to install google's account manager, at least for the time being, side by side with microG replacing other google services. but unfortunately i think this is not possible because microG and its implementation of the google account is monolithic, and i suspect both account managers can't coexist in the same device.

is this suspicion true?

i'd suggest breaking app microG core in two apps to separate the account manager and handle this issue. this would also add the benefit of not installing a google account manager (even if free software) for those who don't want any google service.

**Comment by ale5000-git** [comment]
Personally I'm against splitting the code since it will likely complicate things (like developing and maintining).

I suggest to leave the account code inside GmsCore but inactive and a sort of Google Account Proxy that if present activate it.

Something like "microG Services Framework Proxy".

**Comment by Lanchon** [comment]
yes, this is equally valid, i consider this an implementation detail. this
is probably a good idea (but not familiar with the code, so can't tell).

**Comment by zoff99** [comment]
if those things don't have the same class/package name

**Comment by Lanchon** [comment]
they are split in google's code, so splitting is possible.

**Comment by mar-v-in** [comment]
There is some code splitting stuff in original Play Services but that's not what you want. I might be more work than fixing all open issues reported here and will also increase the size of code unnecessarily.

Since about KitKat, account management is done by play services, GoogleAccountManager.apk is only there for legacy reasons (if it wasn't, Android would loose Google account information on upgrade).

It might be possible to split code into logical units, but that won't allow you to do what you want. And finally: the problems of Hangouts and other apps are merely caused by the account manager but by features missing in microG. Unfortunately the Hangout app is not open source, else it would be easy to find what exactly is problematic and how to fix it. This way it's a lot of work and I have limited time that I'd like to focus on the issues most people care about - and Hangouts is certainly not a feature most free software and privacy enthusiasts care about.

**Comment by Lanchon** [comment]
thanks! of course i understand the priorities. unfortunately, i require hangouts voice in my phone, so no matter how much i like what you are doing here, i'll have to go back to gapps and retry microG during my next phone upgrade. btw, microG works amazingly well, congrats. only google's shitty apps with their undocumented APIs have trouble. thanks again!

**Comment by ale5000-git** [comment]
@mar-v-in: Although it may be a strange way of think; I think that make it working with preinstalled apps like Hangouts and simplify installation will increase popularity between noob users which in turn also increase attention to the project and the possibility to attract developers for helping which will speed up development.

So I think #72 have a lot of importance although not directly for me.
